src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProjectOptionsProcessor.cs

```diff
@@ -62,11 +62,15 @@ internal class ProjectSystemProjectOptionsProcessor : IDisposable
         /// <returns><see langword="true"/> if the command line was updated.</returns>
         private bool ReparseCommandLineIfChanged_NoLock(ImmutableArray<string> arguments)
         {
-            var checksum = Checksum.Create(arguments);
-            if (_commandLineChecksum == checksum)
-                return false;
+            // Ensure reading/writing from the nullable checksum is atomic.
+            lock (this)
```

Comment:
❗ This is already guarded by `_gate` in the caller.

Reply:
removed.